### ISSUE TITLE
Events API does not unsubscribe from events before closing the websocket

### ISSUE THREAD

**Issue by octoquad** [chadxz/awry#139]
Thanks for creating this library!

I've been experimenting with this library using Electron + Vue.js and connecting to ARI to subscribe to all events using `awry.Events.connect()`. I use an Electron event to hook into the window being closed `app.on('before-quit')` to properly close the websocket connection to ARI with `awry.Events.close()`.

This works well, but if you monitor Asterisk after a websocket connection is closed, generally Asterisk will output the following:

```
$ asterisk -rvvv
 Deactivating Stasis app 'ARI-App'
  == WebSocket connection from 'ip.add.re.ss:port' closed
    -- Inactive Stasis app 'ARI-App' missed message
    -- Inactive Stasis app 'ARI-App' missed message
```

The `Inactive Stasis app 'ARI-App' missed message` messages are shown repeatedly after a disconnect, whether the websocket was closed properly or not (no call to `awry.Events.close()`).

Looking at https://github.com/ari4java/ari4java/pull/111/files as a reference, I think the Events API actually needs to unsubscribe before `ReconnectingWebSocket.close()` is called. I'm not 100% sure if this the correct way or if you need to use the `awry.API` in conjuction with `awry.Events`?

I've tried using `awry.API` to get all active subscriptions but either get 404 or 403 if I try and unsubscribe these inactive Stasis apps.

Do you have any pointers with regards to dealing with these Inactive Stasis app messages? If you have many applications connecting to the ARI and don't unsubscribe properly, these messages are interspersed between useful debugging output for system administrators which can be annoying for them.

**Comment by chadxz** [comment]
You've done quite a bit of research here! Are you saying that you tried replicating the behavior you found in the ari4java library in your own application code, but it did not work as you expected? It looks like they are doing the unsubscription prior to disconnecting the websocket. Is that the order of operations in your code?

**Comment by octoquad** [comment]
Well, I'm currently not using `awry.API` which provides `subscribe` and `unsubscribe` functions for certain API's e.g `ApplicationsAPI`. I'm just using `awry.Events` to subscribe to all events  as per https://github.com/chadxz/awry/blob/7c5c718aba4c393f5fb9fc21f22d2726ae68c12d/src/events/index.js#L37

Now from this point on, there is no `subscribe` or `unsubscribe` function in `awry.Events`. There is a `awry.Events.close()` but looking inside of that, there is no `unsubscribe` method being called to unsubscribe before closing the websocket. The reference I provided was whether awry should do the same thing if `subscribeAll` is true (by default), otherwise the onus is on the developer to unsubscribe if they have an explicit subscription request(s), but how do you do that as well?

In all honesty, I'm not sure if `awry.Events.close()` can even unsubscribe before closing the websocket without a massive change, e.g. Register through Applications API and subscribe for `awry.Events.connect()`and unsubcribe through Applications API for `awry.Events.close()`.  

In summary it seems that `awry.Events.connect()` registers a statis app, but `awry.Events.close()` never deregisters it. I am using an old version of Asterisk 13, which is near EOL, so I'm also not sure if this was fixed in later versions of Asterisk and this specific issue is not related to awry at all.

**Comment by chadxz** [comment]
I think what you're experiencing is likely best solved on the [asterisk community forum](https://community.asterisk.org). The asterisk developers are active there and they would be able to explain the behavior you are experiencing.

If it does turn out that something needs to be changed here to provide a better UX, I'm happy to accept a PR. Today though I'm not actively involved in building ARI applications so I'm not tuned into the details needed to help you solve this.

**Comment by octoquad** [comment]
Thanks for the feedback. I am of the opinion that I should actually try this out on a later version of Asterisk to see what happens. If the problem persists, I'll take a look further and see if I can find a solution and provide a PR.
